packages/imperative/src/events/src/EventProcessor.ts

```diff
@@ -0,0 +1,156 @@
+/*
+* This program and the accompanying materials are made available under the terms of the
+* Eclipse Public License v2.0 which accompanies this distribution, and is available at
+* https://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Copyright Contributors to the Zowe Project.
+*
+*/
+
+import { Logger } from "../../logger/src/Logger";
+import { EventCallback, EventTypes } from "./EventConstants";
+import { ImperativeError } from "../../error/src/ImperativeError";
+import { Event } from "./Event";
+import { ConfigUtils } from "../../config/src/ConfigUtils";
+import { LoggerManager } from "../../logger/src/LoggerManager";
+import { ImperativeConfig } from "../../utilities";
+import { EventUtils } from "./EventUtils";
+import { IEventDisposable } from "./doc";
+import { IProcessorTypes } from "./doc/IEventInstanceTypes";
+
+/**
+ * Manages event subscriptions and emissions for a specific application.
+ *
```

Comment:
Elaborate on what it means to the caller to emit an event. Explain why a caller might want to subscribe to an event. Explain that subscribing involves supplying a callback function. Define what a shared event is versus a user event.